cmd/copyEnumeratorInit.go

```diff
@@ -65,7 +67,9 @@ func (cca *cookedCopyCmdArgs) initEnumerator(jobPartOrder common.CopyJobPartOrde
 
 	// Check if the destination is a directory so we can correctly decide where our files land
 	isDestDir := cca.isDestDirectory(cca.destination, &ctx)
-
+	if cca.listOfVersionIDs != nil && (!(cca.fromTo.To() == cca.fromTo.To().Local() || cca.fromTo.To() == cca.fromTo.To().Unknown()) || !isDestDir) {
```

Comment:
In the line 63: `isSourceDir := traverser.isDirectory(true)`, we are checking if source is a directory or not. `isDirectory() ` function internally calls getProperties() which will fail if given source is not a valid blob URL. So we're already checking the source. Checking only destination should do.  So that's why I didn't add check for source. Am I missing something?

Reply:
How does this code behave if I have a filestore as source and a local destination?

Reply:
Rectified the if statement